src/Components/Search/state.ts

```diff
@@ -27,19 +27,28 @@ export class SearchBarState extends Container<StateContainer> {
   reset() {
     this.setState({
       hasEnteredPreviews: false,
-      selectedPreviewIndex: 0,
+      selectedPreviewIndex: null,
       previewItems: [],
     })
   }
 
+  enterPreviewWithoutSelection() {
+    this.setState({ hasEnteredPreviews: true })
+  }
+
   enterPreview() {
     this.setState({ hasEnteredPreviews: true, selectedPreviewIndex: 0 })
   }
 
+  leavePreviewIfNoSelection() {
```

Comment:
When leaving with the mouse, you only want to set the flag to false if you're not already in there from the keyboard.

Reply:
Given this helpful note, would it make sense to rename `selectedPreviewIndex` to indicate that it was set by keyboard use? Maybe `previewIndexSelectedByKeyboard` or something.